# **Exercise 1 (Cutlets data exercise)**

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

 **Assume**,

 Null hypothesis h0 = Diameters of the cutlets of the two units are similar

 Alternate hypothesis ha = Diameters of the cutlets of the two units are dissimilar

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving Cutlets.csv to Cutlets.csv


In [ ]:
data=pd.read_csv("Cutlets.csv")
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
#let mu1 be the mean of cutlets of unit A and mu2 be the mean of cutlets of unit B
#Hence h0 : mu1 - mu2 = 0 & ha: mu1 - mu2 != 0
unit_A = data['Unit A']
unit_B = data['Unit B']

In [ ]:
#Since this is a sample we have to use t test
stats.ttest_ind(unit_A , unit_B)


TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

**Conclusion**:Here significence level is 5% so alpha is 0.05

p value ~ 0.4722 which is greater than alpha which means that the diameters of the cutlets of the two units are not significantly different

If p value is not equal to  α we failed to reject Null Hypothesis because of lack of evidence

# **Exercise 2 (LabTAT data exercise)**

Here we have 4 different types laboratory so let their respective means be mu1,mu2,mu3,mu4

**Assume**,
Null hypothesis h0 = all the means are equal i.e mu1=mu2=mu3=mu4.

Alternative hypothesis ha = atleast one of the mean is different.

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving LabTAT.csv to LabTAT.csv


In [ ]:
df=pd.read_csv("LabTAT.csv")
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
df.shape

(120, 4)

In [ ]:
#Since we have to find if the means of 4 columns are equal or not, we have to use f test or anova test
stats.f_oneway(df.iloc[:,0], df.iloc[:,1],df.iloc[:,2],df.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

**Conclusion:** Here p value is almost nearly zero and is thus less than alpha or significance level i.e 5% or 0.05.
This means that we reject h0 and accept ha.

Hence, there is indeed a difference in the average TAT values between the 4 laboratories

# **Exercise 3 (BuyerRatio data exercise)**

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving BuyerRatio.csv to BuyerRatio (1).csv


In [ ]:
df1 = pd.read_csv('BuyerRatio.csv')
df1.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

**Assume**, h0 = Proportions are equal and ha = Proportions are not equal

In [ ]:
Data = df1.iloc[:,1:5]

In [ ]:
Data

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [ ]:
#Since there are counts of males and females across different regions, we use chi square to find relation between them.


In [ ]:
stats.chi2_contingency(Data)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

**Conclusion:** Here the p value is 0.6603 which is greater than 0.05 i.e alpha (significance level). Hence proportions are not equal.

This means that we reject h0 and accept ha.

# **Exercise 4 (Costomer+OrderForm data exercise)**

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [ ]:
df2 = pd.read_csv('Costomer+OrderForm.csv')


In [ ]:
df2.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

**Assume**, h0 = defective% does not vary by center & ha = defective% does vary by center with alpha being 0.05
     


In [ ]:
df2['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
df2['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
df2['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
df2['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
observed = ([[271,276,269,280],[29,33,31,20]])

In [ ]:
observed

[[271, 276, 269, 280], [29, 33, 31, 20]]

In [ ]:
stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

**Conclusion:** Here p value is 0.2771 which is greater than alpha i.e 0.05 and thus the defectice % does vary by center.

This means that we reject h0 and accept ha.

